frontend/src/components/common/Resource/EditorDialog.tsx

```diff
@@ -557,3 +569,226 @@ export default function EditorDialog(props: EditorDialogProps) {
 export function ViewDialog(props: Omit<EditorDialogProps, 'onSave'>) {
   return <EditorDialog {...props} onSave={null} />;
 }
+
+interface UploadYamlProps {
+  setUploadYaml: (value: boolean) => void;
+  setCode: React.Dispatch<React.SetStateAction<{ code: string; format: string }>>;
+}
+
+function UploadYourYaml(props: UploadYamlProps) {
+  const { setUploadYaml, setCode } = props;
+  const { t } = useTranslation();
+  const [uploadedfile, setUploadedFile] = React.useState<File | null>(null);
+
+  const ActionButtons = ({ onBack, onLoad }: { onBack: () => void; onLoad: () => void }) => (
+    <Box sx={{ display: 'flex', mt: 2, justifyContent: 'space-between' }}>
+      <Button variant="outlined" sx={{ borderColor: 'grey.500', px: 2 }} onClick={onBack}>
+        <Icon
+          icon="mdi:arrow-left"
+          width="18"
+          height="18"
+          style={{ display: 'inline-block', marginRight: '8px' }}
+        />
+        {t('translation|Back')}
+      </Button>
+      <Button
+        onClick={onLoad}
+        variant="contained"
+        sx={{
+          bgcolor: 'grey.700',
```

Comment:
please don't use hardcoded colors
buttons can be `color="primary"` or `color="secondary"`